### PR TITLE
Add two Sendable annotations to enable building SourceKit-LSP in Swift 6 mode

### PR DESCRIPTION
**Pull request by ahoppen**
* **Explanation**: Mark `FileRuleDescription` and `Platform` as `Sendable` and make `Platform.current` a constant. This allows us to build SourceKit-LSP in Swift 6 mode
* **Scope**: `FileRuleDescription` and `Platform` if SwiftPM is used as a library
* **Risk**: Very low, adding `Sendable` conformances is safe and I don’t expect anyone to modify `Platform.current`
* **Testing**: Verified that this helps us build SourceKit-LSP in Swift 6 mode
* **Issue**: n/a
* **Reviewer**:  @xedin on https://github.com/apple/swift-package-manager/pull/7553

### CODE REVIEW

File: Sources/PackageLoading/Platform.swift [hunk 1]
```diff
@@ -20,23 +20,21 @@ private func isAndroid() -> Bool {
...
-    public enum LinuxFlavor: Equatable {
+    public enum LinuxFlavor: Equatable, Sendable {
         case debian
         case fedora
     }
 }
 
 extension Platform {
-    // This is not just a computed property because the ToolchainRegistryTests
-    // change the value.
```
Comment (bnbarham) [inline Sources/PackageLoading/Platform.swift:38]: I assume this comment wasn't true 😅? If that's the case though, can it just be a computed property?

**Review by bnbarham — APPROVED** [review]

File: Sources/PackageLoading/Platform.swift (hunk 1 above)
Comment (ahoppen) [reply Sources/PackageLoading/Platform.swift:38]: It was, `ToolchainRegistryTests` is in SourceKit-LSP: https://github.com/apple/sourcekit-lsp/pull/1276#discussion_r1597909530

I prefer it to be stored so we don’t re-run `uname` every time the variable is accessed.

File: Sources/PackageLoading/TargetSourcesBuilder.swift [hunk 2]
```diff
@@ -563,7 +563,7 @@ public struct TargetSourcesBuilder {
 }
 
 /// Describes a rule for including a source or resource file in a target.
-public struct FileRuleDescription {
+public struct FileRuleDescription: Sendable {
```
Comment (giginet) [inline Sources/PackageLoading/TargetSourcesBuilder.swift:566]: Don't we need `FileRuleDescription.Rule` to conform to `Sendable`?

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Linux

**Pull request merged**